x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToBooleanFromDoubleEvaluator.java

```diff
@@ -58,9 +63,9 @@ public Block evalVector(Vector v) {
       }
     }
     return nullsMask == null
-          ? new BooleanArrayVector(values, positionCount).asBlock()
+          ? new BooleanArrayVector(values, positionCount, driverContext.blockFactory()).asBlock()
           // UNORDERED, since whatever ordering there is, it isn't necessarily preserved
-          : new BooleanArrayBlock(values, positionCount, null, nullsMask, Block.MvOrdering.UNORDERED);
+          : new BooleanArrayBlock(values, positionCount, null, nullsMask, Block.MvOrdering.UNORDERED, driverContext.blockFactory());
```

Comment:
Hmm.. we need to be careful here. This constructor assumes that the breaker accountancy has been done. I think what you want is:
 blockFactory.newBooleanArrayBlock(..)

And yes... this is a little error prone. We'll restrict access to these constructors once we're further along.